drivers/sensor/bmi08x/bmi08x_accel.c

```diff
@@ -0,0 +1,778 @@
+{
+	int32_t converted_val;
+
+	/*
+	 * maximum converted value we can get is: max(raw_val) * max(scale)
+	 *	max(raw_val) = +/- 2^15
+	 *	max(scale) = 4785
+	 *	max(converted_val) = 156794880 which is less than 2^31
+	 */
+	converted_val = raw_val * scale;
+	val->val1 = converted_val / 1000000;
+	val->val2 = converted_val % 1000000;
+}
+
+static void bmi08x_channel_convert(enum sensor_channel chan, uint16_t scale, uint16_t *raw_xyz,
+				   struct sensor_value *val)
+{
+	int i;
+	uint8_t ofs_start, ofs_stop;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		ofs_start = ofs_stop = 0U;
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		ofs_start = ofs_stop = 1U;
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		ofs_start = ofs_stop = 2U;
+		break;
+	default:
+		ofs_start = 0U;
+		ofs_stop = 2U;
+		break;
+	}
+
+	for (i = ofs_start; i <= ofs_stop; i++, val++) {
+		bmi08x_to_fixed_point(raw_xyz[i], scale, val);
+	}
+}
+
+static inline void bmi08x_acc_channel_get(const struct device *dev, enum sensor_channel chan,
+					  struct sensor_value *val)
+{
+	struct bmi08x_accel_data *data = dev->data;
+
+	bmi08x_channel_convert(chan, data->scale, data->acc_sample, val);
+}
+
+static int bmi08x_temp_channel_get(const struct device *dev, struct sensor_value *val)
+{
+	uint16_t temp_raw = 0U;
+	int32_t temp_micro = 0;
+
+	if (bmi08x_accel_word_read(dev, BMI08X_REG_TEMP_MSB, &temp_raw) < 0) {
+		return -EIO;
+	}
+
+	/* the scale is 1/2^5/LSB = 31250 micro degrees */
+	temp_micro = BMI08X_TEMP_OFFSET * 1000000ULL + temp_raw * 31250ULL;
+
+	val->val1 = temp_micro / 1000000ULL;
+	val->val2 = temp_micro % 1000000ULL;
+
+	return 0;
+}
+
+static int bmi08x_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	(void)pm_device_state_get(dev, &state);
+	if (state != PM_DEVICE_STATE_ACTIVE) {
+		return -EBUSY;
+	}
+#endif
+
+	switch ((int16_t)chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		bmi08x_acc_channel_get(dev, chan, val);
+		return 0;
+	case SENSOR_CHAN_DIE_TEMP:
+		return bmi08x_temp_channel_get(dev, val);
+	default:
+		LOG_ERR("Channel not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int bmi08x_accel_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	uint8_t conf_reg_val;
+	uint8_t ctrl_reg_val;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		conf_reg_val = BMI08X_ACCEL_PM_ACTIVE;
+		ctrl_reg_val = BMI08X_ACCEL_POWER_ENABLE;
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		conf_reg_val = BMI08X_ACCEL_PM_SUSPEND;
+		ctrl_reg_val = BMI08X_ACCEL_POWER_DISABLE;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if (bmi08x_accel_byte_write(dev, BMI08X_REG_ACCEL_PWR_CONF, conf_reg_val) < 0) {
+		LOG_ERR("Failed to set conf power mode");
+		return -EIO;
```

Comment:
Is there any particular reason why you are not propagating the return codes from `_read/_write` functions?

Reply:
I honstely can't remember why by this point... I completed this code 2 years ago and only just opened a PR in October...

I fixed it in both the accel and gyro code